src/patients/related-persons/AddRelatedPersonModal.tsx

```diff
@@ -45,15 +47,30 @@ const AddRelatedPersonModal = (props: Props) => {
 
   const onSearch = async (query: string) => {
     const patients: Patient[] = await PatientRepository.search(query)
-    return patients.filter((p: Patient) => p.id !== patient.id)
+    return patients.filter((p: Patient) => p.id !== patientId)
+  }
+
+  const onSaveButtonClick = async () => {
+    try {
+      await mutate({ patientId, relatedPerson })
+      onCloseButtonClick()
+    } catch (e) {
+      setRelatedPersonError(e)
+    }
   }
 
   const formattedDate = (date: string) => (date ? format(new Date(date), 'yyyy-MM-dd') : '')
 
   const body = (
     <form>
-      {relatedPersonError?.message && (
-        <Alert color="danger" title={t('states.error')} message={t(relatedPersonError?.message)} />
+      {relatedPersonError && (
+        <Alert
+          color="danger"
+          title={t('states.error')}
+          message={t(
+            relatedPersonError.relatedPersonError || relatedPersonError.relationshipTypeError,
```

Comment:
This should be a generic "Could not add related person message". This error should not contain any information about a specific field.